src/vsphere_cpi/lib/nsxt/nsxt_client/api_client.rb

```diff
@@ -48,125 +38,94 @@ def self.default
     # @return [Array<(Object, Fixnum, Hash)>] an array of 3 elements:
     #   the data deserialized from response body (could be nil), response status code and response headers.
     def call_api(http_method, path, opts = {})
-      request = build_request(http_method, path, opts)
-      response = request.run
+      http_method = http_method.to_sym.downcase
 
+      request_body = build_request_body(opts[:body])
+
+      if @config.debugging
+        @config.logger.debug "HTTP request body param ~BEGIN~\n#{request_body}\n~END~\n"
+      end
+      request = authentication.headers(default_headers)
+      request_url = build_request_url(path)
+      if has_proxy(request_url)
+        request = request.via(*proxy(request_url))
+      end
+      response = request.send(http_method,
+                              request_url,
+                              params: opts[:query_params],
+                              body: request_body,
+                              ssl_context: ssl_context)
       if @config.debugging
         @config.logger.debug "HTTP response body ~BEGIN~\n#{response.body}\n~END~\n"
       end
 
-      unless response.success?
-        if response.timed_out?
-          fail ApiCallError.new('Connection timed out')
-        elsif response.code == 0
-          # Errors from libcurl will be made visible here
-          fail ApiCallError.new(:code => 0,
-                            :message => response.return_message)
-        else
-          api_error = ApiError.new(JSON.parse(response.body))
-          error_message = api_error.related_errors ? api_error.related_errors : api_error.error_message
-          fail ApiCallError.new(:code => response.code,
-                            :response_headers => response.headers,
-                            :response_body => response.body,
-                            :message => error_message),
-               response.status_message
-        end
+      unless has_good_response_code?(response.code)
+        api_error = ApiError.new(JSON.parse(response.body))
+        error_message = api_error.related_errors ? api_error.related_errors : api_error.error_message
+        fail ApiCallError.new(:code => response.code,
+                              :response_headers => response.headers,
+                              :response_body => response.body,
+                              :message => error_message),
+             response.status
       end
 
       if opts[:return_type]
-        data = deserialize(response, opts[:return_type])
+        data = deserialize(response.to_s, opts[:return_type])
```

Comment:
why are we doing to_s on response? should this be response.body

Reply:
`response.to_s` gives you a response body.